### DISCUSSION TITLE
PID using two interlocking relays to control 3T valve

### DISCUSSION THREAD

**Discussion by degreal** [arendst/Tasmota#14893] [Ideas]
I have the same problem described in here https://github.com/colinl/Sonoff-Tasmota/issues/8 but I understand all this functionality is already merged?

How do you solve shutter drift issue? I calibrated 3T valve as a shutter precisely as I can but in a week it's going to drift so it will not fully close or fully open, you have to cycle from 0% -to 100% so that the position in Tasmota reflects the reality. Often times it's impossible to cycle it to 100% because 110c water is going to melt my pex pipes.

**Comment by stefanbode** [comment]
Hi, on the pure time based approach a drift cannot be avoided completely. Therefore most users use at least one of the end-positions to "recalibrate" and get the shift eliminated. The only workaround is e.g. a STEPPER motor. These are position based and have clear positions. Works e.g. excellent with moving doors. There is no shift and position is even after years correct. Do you have an idea how to overcome the issue? Maybe we can code it.

**Comment by degreal** [comment]
This PID works great with single relay, example heating a boiler. Am I completely oblivious when I think this is exactly the same thing as heating a boiler but instead of just having "heating" relay we also have "cooling" relay? When temperature gets too hot then instead of waiting boiler to cool down we speed up the process by enabling the "cooling" relay?
